src/cache.rs

```diff
@@ -144,20 +170,39 @@ impl Cache {
 
     /// Check for pages for a given platform in one of the given languages.
     fn find_page_for_platform(
-        name: &str,
+        page_name: &str,
         cache_dir: &Path,
         platform: &str,
         language_dirs: &[String],
-    ) -> Option<PathBuf> {
+    ) -> Option<PageLookupResult> {
```

Comment:
In my opinion, this now should not return a `PageLookupResult` anymore, because the result of this function will never "exhaust the full potential of the type" / will always have patch be `None` and therefore makes handling the result more complex than it has to be.